gcc/rust/checks/errors/borrowck/polonius/rust-polonius.h

```diff
@@ -45,7 +45,7 @@ struct FullPoint
   static uint32_t extract_bb (Point point) { return point >> 16; }
   static uint32_t extract_stmt (Point point)
   {
-    return (point & ~(1 << 16)) >> 1;
+    return (point >> 1) & ((1 << 15) - 1);
```

Comment:
To be fair I would have not written it like this. We lack any documentation/intent hint with such expression. In such case I would:
- Use many constexpr or define to describe the intent of the operation (What does the 1 correspond to ? What about the mask ?) We should give a name to every hard coded constant.
- I would also (probably) change `((1 << 15) - 1)` to `0x7fff`, I think it better hints about the binary representation ? To be fair I'd like to hear from you about this.